Class/Practise 5.1/src/MegaDataPool.cpp

```diff
@@ -0,0 +1,42 @@
+#include "MegaDataPool.hpp"
+#include <stdexcept>
+
+MegaDataPool::MegaDataPool(size_t poolSize)
+    : poolSize(poolSize), pool(poolSize), used(poolSize, false) {
+    for (size_t i = 0; i < poolSize; ++i) {
+        pool[i] = std::make_shared<MegaData>();
+    }
+}
+
+std::shared_ptr<MegaData> MegaDataPool::acquire() {
+    for (size_t i = 0; i < poolSize; ++i) {
+        if (!used[i]) {
```

Comment:
Maybe we may need some mutex to support multithreaded case too.

Reply:
I added **std::lock_guard<std::mutex>**  to lock the mutex in a scoped manner, automatically unlocking when the scope ends.